Update express checkouts section in settings for mobile view

naman03malhotra opened the pull request:
Fixes #6428 

#### Changes proposed in this Pull Request

- Aligned mobile view for express checkouts - apple pay and stripe link, woopay

Please refer to [this design](C1r0Vzxh1mhbDJJnbjHp5q-fi-2135_82787) for expected design

### Layout 1
| Before | After |
|--------|--------|
| ![image (2)](https://github.com/Automattic/woocommerce-payments/assets/110893979/b09def23-69b2-402d-a82a-c49c9956337a) | ![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/3130f8b8-a4c2-4ba8-8d2d-52a76125a167) |

| Before | After |
|--------|--------|
| ![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/1a34336b-0283-4984-8aa6-26f962b4a802) | ![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/105a4128-c35b-4d0f-8e5a-d94dba71b427) |


![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/3130f8b8-a4c2-4ba8-8d2d-52a76125a167)

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Goto Payment > Settings
- Enable Multiple payment methods by switching store currency to EUR
- Open the browser dev tools and switch to the mobile view
- Check mobile responsiveness as expected
- Make sure the desktop version UI doesn't change and is as expected.
- Check the dropdown position in both mobile and desktop versions.
- Check if the functionality is intact and doesn't change on either desktop or mobile.


![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/f09d4286-8431-4db2-bfdc-c7860df04c57)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

naman03malhotra committed:
Updated express checkouts section in settings so that it is rendered correctly in mobile

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6984` or branch name `imp/6428-improve-settings-mobile-view-express-checkouts` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=imp/6428-improve-settings-mobile-view-express-checkouts) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f67f7d399d812141028459c72efc6da028b3725c
- Build time: 2023-08-23 13:01:12 UTC

_Note: the build is updated when a new commit is pushed to this PR._

pierorocca commented:
@elizaan36 could you double check this is as expected on mobile? With the narrower size, the text looks a bit odd.

naman03malhotra committed:
Improved design of express checkout settings Page in mobile view - WooPay

naman03malhotra commented:
>could you double check this is as expected on mobile? With the narrower size, the text looks a bit odd.

@pierorocca, maybe the screenshot was not clear, I've added a new one in the description. If something is still odd, please let me know.


![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/50110e3e-09f5-4465-ace7-e13071fb259f)

naman03malhotra committed:
Merge branch 'develop' into imp/6428-improve-settings-mobile-view-express-checkouts

pierorocca commented:
The formatting looks odd to me now that the sentence space is shorter. It looks choppy and the copy for Apple and Google could fit in 3 lines instead of 4. What would it look like without the line break?

naman03malhotra committed:
removed log file

naman03malhotra committed:
Merge branch 'develop' into imp/6428-improve-settings-mobile-view-express-checkouts

elizaan36 commented:
> @pierorocca, maybe the screenshot was not clear, I've added a new one in the description. If something is still odd, please let me know.

Yup, looks like there's a hard line break where there shouldn't be. The line "By enabling..." should wrap with the rest of the text.

naman03malhotra commented:
I have added removed the new line and added some spacing in the paragraph.
![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/684d51ea-83f0-4be5-afec-b4814ea1cfb1)

@pierorocca @elizaan36 please check

naman03malhotra committed:
Merge branch 'develop' into imp/6428-improve-settings-mobile-view-express-checkouts

naman03malhotra committed:
added spacing and removed new line

pierorocca commented:
Yes so much better than you! Could we see the checked state please?

naman03malhotra committed:
Merge branch 'develop' into imp/6428-improve-settings-mobile-view-express-checkouts

naman03malhotra commented:
Here you go @pierorocca 
![image](https://github.com/Automattic/woocommerce-payments/assets/15019298/ae10891c-5f2a-4976-88f9-3fba3631a30a)

pierorocca commented:
Thanks @naman03malhotra. Looks like a huge improvement overall on Mobile. Thank you for making WooPayments better. LTGM.

htdat commented on changelog/imp-6428-improve-settings-mobile-view-express-checkouts:
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: update
+
+Updated express checkouts section in settings so that it is rendered correctly in mobile
```
Nitpick: 

```suggestion
Update express checkouts section in settings for mobile view
```

htdat committed:
Merge branch 'develop' into imp/6428-improve-settings-mobile-view-express-checkouts

htdat commented on client/settings/express-checkout/apple-google-pay-item.tsx:
```diff
@@ -0,0 +1,167 @@
+							<div className="express-checkout__icon">
+								<img src={ ApplePay } alt="Apple Pay" />
+							</div>
+							<div className="express-checkout__label express-checkout__label-mobile">
+								{ __( 'Apple Pay', 'woocommerce-payments' ) }
+							</div>
```
Right below, there is another similar div with only difference is the classname `desktop`. 

It might be worth extracting it to a new component with two props "title" and "viewType". With that, it can be reused 8 times across express checkout components here.

htdat reviewed (commented):
@naman03malhotra - As mentioned in another review https://github.com/Automattic/woocommerce-payments/pull/6903#discussion_r1289826041 (before you splitted this PR), the technical approach looks OK. 

Besides inline comments, could you use this opportunity to convert components of "Link" and "WooPay" to TypeScript too, which is similar to what you've done for "Apple Pay / Google Pay"?

naman03malhotra committed:
Update changelog/imp-6428-improve-settings-mobile-view-express-checkouts

Co-authored-by: Dat Hoang <htdat@users.noreply.github.com>

naman03malhotra committed:
Merge branch 'develop' into imp/6428-improve-settings-mobile-view-express-checkouts

naman03malhotra commented:
Thanks for the review Dat.
>Besides inline comments, could you use this opportunity to convert components of "Link" and "WooPay" to TypeScript too, which is similar to what you've done for "Apple Pay / Google Pay"?

The reason I didn't do it as part of this PR is as it required some changes to be done in tsconfig and I would like to handle it in a separate issue. Created here: https://github.com/Automattic/woocommerce-payments/issues/7052

>It might be worth extracting it to a new component with two props "title" and "viewType". With that, it can be reused 8 times across express checkout components here.

Thanks for pointing this out Dat, though this is a good improvement, but I would like to go a step further and extract the row component and while solving that issue I plan to extract the title component. Created issue here: https://github.com/Automattic/woocommerce-payments/issues/7053

htdat reviewed (approved):
Thanks for your answer. We can handle my questions in other issues.

LGTM, and this is ready to merge!
